### PR TITLE
Add a test for non-repo error message

### PR DESCRIPTION
**Pull request by eljobe**
Fixes gradle/gradle-native#213

### CODE REVIEW

File: subprojects/version-control/src/test/groovy/org/gradle/vcs/git/internal/GitVersionControlSystemSpec.groovy [hunk 1]
```diff
@@ -150,6 +150,23 @@ class GitVersionControlSystemSpec extends Specification {
         thrown GradleException
     }
 
+    def 'error if repo is not a git repo'() {
+        given:
+        def target = tmpDir.file('versionDir')
+        repoSpec.url = 'https://notarepo.io'
```
Comment (big-guy) [inline subprojects/version-control/src/test/groovy/org/gradle/vcs/git/internal/GitVersionControlSystemSpec.groovy]: We should use [`.invalid`](https://en.wikipedia.org/wiki/.invalid) for this or another one of the test domain names

Comment (eljobe) [reply subprojects/version-control/src/test/groovy/org/gradle/vcs/git/internal/GitVersionControlSystemSpec.groovy]: Done.

**Comment by big-guy** [comment]
@eljobe I tried this locally with an integration test (`GitVcsIntegrationTest`):
```

    def 'reasonable error when VCS url is incorrect'() {
        given:
        settingsFile << """
            sourceControl {
                vcsMappings {
                    withModule("org.test:dep") {
                        from vcs(GitVersionControlSpec) {
                            url = "https://notarepo.invalid"
                        }
                    }
                }
            }
        """
        expect:
        fails("assemble")
    }
```

This is the kind of message I see:

> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> > Could not determine the dependencies of task ':compileJava'.
> > > Could not resolve all dependencies for configuration ':compileClasspath'.
> > > > Exception caught during execution of ls-remote command
> 
> * Try:
> Run with --info or --debug option to get more log output.
> ... (long stacktrace) ...
> 
> * Get more help at https://help.gradle.org
> 
> BUILD FAILED in 6s

I think it would be better if the underlying cause was a bit more user-level (instead of referring to "ls-remote"), e.g. maybe something like:

> * What went wrong:
> > Could not determine the dependencies of task ':compileJava'.
> > > Could not resolve all dependencies for configuration ':compileClasspath'.
> > > > Could not list available versions from https://notarepo.invalid

The underlying JGit exceptions can still be the cause, we just won't see that unless someone runs with `-s`.

WDYT?

**Comment by eljobe** [comment]
Okay. Take another look at let me know what you think of it.

**Pull request merged**